src/DAL/ImageDataRepository.ts

```diff
@@ -0,0 +1,53 @@
+import { Repository, EntityRepository } from 'typeorm';
+import { container, delay, inject } from 'tsyringe';
+import { MCLogger } from '@map-colonies/mc-logger';
+import { ImageData } from '../entity/ImageData';
+
+@EntityRepository(ImageData)
+export class ImageDataRepository extends Repository<ImageData> {
+
+  private readonly mcLogger: MCLogger; //don't override internal repository logger.
+
+  public constructor(
+  ) {
+    super();
+    this.mcLogger = container.resolve(MCLogger); //direct injection don't work here due to being initialized by typeOrm
+  }
+
+  public async get(id: string): Promise<ImageData | undefined> {
+    return this.findOne({ id: id });
+  }
+
+  public async createAndSave(image: ImageData): Promise<void> {
+    const exists = (await this.get(image.id)) != undefined;
+    if (!exists) {
+      await this.save(image);
+    } else {
+      this.mcLogger.info('duplicate value error has occurred');
```

Comment:
add data to error log

Reply:
error handling here is temporary will be handled in separate task/pr